### ISSUE TITLE
Allow to dispatch an action on async form validation

### ISSUE THREAD

**Issue by yoo2001818** [erikras/redux-form#73]
redux-form doesn't seem to allow to dispatch an action on async validation for now.
However, I seperated ajax logic with async action and middleware like this:
```js
const fetch = createAction('SEARCH_FETCH', 
  (query) => api('GET', '/api/search', {query}));

// This returns a Promise for searching 'Hello, world!'
// and this also dispatches 'SEARCH_FETCH' when fetching is complete.
store.dispatch(fetch('Hello, world!'));
.then(result => {
  console.log(result);
});
```

If redux-form allows to dispatch an (async) action on form validation, I think it can seperate API call logic and validation logic.
```js
const checkUsername = createAction('SIGNUP_CHECK_USERNAME', 
  (username) => api('GET', '/api/signup/checkUsername', {username}));
function validateFormAsync(data, dispatch) {
  // In this case, checkUsername action is ignored by reducers
  return dispatch(checkUsername(data.username));
  .then(action => {
    if (action.error) {
      return {
        username: 'Internal server error'
      };
    }
    return {
     username: action.payload
    };
  });
}
```

Here's the source code of 'api' function: https://gist.github.com/yoo2001818/05aad13f1af7fb5eaf38

**Comment by erikras** [comment]
Sorry, but I'm not understanding exactly what you're asking/suggesting. You just want `dispatch` to be given to the `asyncValidate` function?

**Comment by yoo2001818** [comment]
Basically, yes.

**Comment by erikras** [comment]
Fixed in [v1.6.0](https://github.com/erikras/redux-form/releases/tag/v1.6.0).
